metadata-ingestion/src/datahub/ingestion/source/unity/ge_profiler.py

```diff
@@ -0,0 +1,170 @@
+import logging
+from concurrent.futures import ThreadPoolExecutor, as_completed
+from dataclasses import dataclass, field
+from typing import Iterable, List, Optional
+
+from sqlalchemy import create_engine
+from sqlalchemy.engine import Connection
+
+from datahub.ingestion.api.workunit import MetadataWorkUnit
+from datahub.ingestion.source.sql.sql_config import SQLCommonConfig
+from datahub.ingestion.source.sql.sql_generic import BaseTable
+from datahub.ingestion.source.sql.sql_generic_profiler import (
+    GenericProfiler,
+    TableProfilerRequest,
+)
+from datahub.ingestion.source.unity.config import UnityCatalogGEProfilerConfig
+from datahub.ingestion.source.unity.proxy_types import Table, TableReference
+from datahub.ingestion.source.unity.report import UnityCatalogReport
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass(init=False)
+class UnityCatalogSQLGenericTable(BaseTable):
+    ref: TableReference = field(init=False)
+
+    def __init__(self, table: Table):
+        self.name = table.name
+        self.comment = table.comment
+        self.created = table.created_at
+        self.last_altered = table.updated_at
+        self.column_count = len(table.columns)
+        self.ref = table.ref
+        self.size_in_bytes = None
+        self.rows_count = None
+        self.ddl = None
+
+
+class UnityCatalogGEProfiler(GenericProfiler):
```

Comment:
the amount of duplicated code here is not ideal - we definitely will need to revisit the GE profiler at some point

Reply:
Yeah, ideally I'd merge this with Mayuri's changes but if I don't get this in now idk when it'll go in... don't really have time to iterate on this